presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteConstantArrayContainsToInExpression.java

```diff
@@ -100,7 +100,7 @@ private boolean isSupportedType(Type type)
         @Override
         public RowExpression rewriteCall(CallExpression node, Void context, RowExpressionTreeRewriter<Void> treeRewriter)
         {
-            if (functionResolution.isArrayContainsFunction(node.getFunctionHandle()) && node.getArguments().get(0) instanceof ConstantExpression) {
+            if (functionResolution.isArrayContainsFunction(node.getFunctionHandle()) && node.getArguments().get(0) instanceof ConstantExpression && !((ConstantExpression) node.getArguments().get(0)).isNull()) {
```

Comment:
if array is NULL should we rewrite this to replace the whole expression by NULL?
```
presto:tpch> select contains(null, 1);
 _col0
-------
 NULL
(1 row)

```

Reply:
Just checked, this is already done by our optimizer (possibly by RowExpressionInterpreter) and is out of the scope of this optimizer.